src/protocols/MeterW1therm.cpp

```diff
@@ -83,8 +83,50 @@ bool MeterW1therm::W1sysHWif::readTemp(const std::string &device, double &value)
 			print(log_debug, "CRC not ok from %s (%s)", "w1t", dev.c_str(), buffer);
 		} else {
 			// crc ok
-			// now parse t=<value>
+			// now check for other errors:
 			if (fgets(buffer, 100, fp)) { // e.g. 07 01 55 00 7f ff 0c 10 18 t=16437
+				
+				///// Start of new code: Catch DS18B20-specific errors.       /////
```

Comment:
maybe put this into a function to isolate it, instead of surrounding with comments?
`private check_quirks(device,buffer){...`
`if (!check_quirks(device,buffer)) return false;`